File: asd3 inwersje.c
#include <stdio.h>
#include <stdlib.h>

int main()
{
 // program sprawdza ile razy zachodzi inwersja polegaj¹ca na tym ¿e liczba x>y  i wartoœæ x<y

    int i, k, j,g, p, inw;
    int t[10000];
    scanf("%d",&i);
    while(i--) // wykonuje siê podan¹ w i liczbê razy
    {
        inw=0;  //licznik inwersji, zerowany przy ka¿dym nowym zestawie danych
        scanf("%d",&p);  //iloœæ liczb w danym zestawie
        for(j=0;j<p;j++) //wykonuje siê o 1 mniej ni¿ p bo zapisuje w tablicy w której pierwsze miejsce jest zerem
        {
            scanf("%d",&t[j]); // zapisuje podane po spoacji liczby
        }

        for(g=0;g<p;g++) //bêdzie ona iteratorem liczby sprawdzaj¹cej
        {
            for(j=g;j<p;j++) // iterator liczby sprawdzanej
            {
                if(t[g]>t[j]) // jeœli liczba sprawdzaj¹ca jest wiêksza ni¿ sprawdzana która jest od niej dalej w tablicy to zachodzi inwersja
                    inw++;
            }
        }
            printf("%d\n",inw);
    }
    return 0;


}

File: asd3 inwersje.py
i=int(input()) #liczba zestawów danych
while i>0 :
    l=int(input()) #ilość podanych liczb w zestawie
    a=[]
    a=[int(s) for s in input().split()] # lista w której zawierają się podane po spacji liczby
    j=l-1 # -1 bo ta wartość jest używana w pętli i tak mi bardziej pasowało
    licz=0 #licznik inwersji
    k = 0 #numer na liście liczby sprawdzającej
    while j>=0 :
        m=j
        while m > 0 :
            if a[k]>a[k+m]: # np a[0]>a[0+5] piątka to ilość liczb w zestawie , porównuje jedną liczbą z pozostałymi z listy od prawej strony
                licz=licz+1
            m-=1
        k+=1
        j-=1
    i-=1
    print(licz)

